src/tests/subcommands/add_test.go

```diff
@@ -0,0 +1,104 @@
+package tests
+
+import (
+	"bytes"
+	"encoding/gob"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"accretional.com/semantifly/subcommands"
+)
+
+func createTempFile(t *testing.T, dir string, data []byte) *os.File {
+	file, err := os.CreateTemp(dir, "testfile")
+	if err != nil {
+		t.Fatalf("Failed to create temp file: %v", err)
+	}
+	_, err = file.Write(data)
+	if err != nil {
+		t.Fatalf("Failed to write to temp file: %v", err)
+	}
+	err = file.Close()
+	if err != nil {
+		t.Fatalf("Failed to close temp file: %v", err)
+	}
+	return file
+}
+
+func createTempDir(t *testing.T) string {
+	dir, err := os.MkdirTemp("", "testdir")
+	if err != nil {
+		t.Fatalf("Failed to create temp directory: %v", err)
+	}
+	return dir
+}
+
+func TestCopyFile(t *testing.T) {
```

Comment:
maybe make the name TestReadWrite instead

Reply:
Changed it.